tests/ui/cross-crate/cross-crate-refcell-match.rs

```diff
@@ -0,0 +1,39 @@
+//! Regression test for https://github.com/rust-lang/rust/issues/16822
+//
+//! ICE when using RefCell::borrow_mut()
+//! inside match statement with cross-crate generics.
+//!
+//! The bug occurred when:
+//! - A library defines a generic struct with RefCell<T> and uses borrow_mut() in match
+//! - Main crate implements the library trait for its own type
+//! - Cross-crate generic constraint causes type inference issues
+//!
+//! The problematic match statement is in the auxiliary file, this file triggers it.
+//! Original error: assertion failed in ty.rs:2481 during type inference.
```

Comment:
Nit: drop this "original error", it would have no correspondence with current compiler impl.